test/new-e2e/tests/apm/vm_test.go

```diff
@@ -0,0 +1,185 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package apm
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+type VMFakeintakeSuite struct {
+	e2e.BaseSuite[environments.Host]
+	transport transport
+}
+
+func vmSuiteOpts(tr transport, opts ...awshost.ProvisionerOption) []e2e.SuiteOption {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+	opts = append(opts, awshost.WithDocker())
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awshost.Provisioner(opts...)),
+		e2e.WithStackName(fmt.Sprintf("apm-vm-suite-%s-%v", tr, os.Getenv("CI_PIPELINE_ID"))),
+	}
+	return options
+}
+
+// TestVMFakeintakeSuiteUDS runs basic Trace Agent tests over the UDS transport
+func TestVMFakeintakeSuiteUDS(t *testing.T) {
+	cfg := `
+apm_config.enabled: true
+apm_config.receiver_socket: /var/run/datadog/apm.socket
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Not a blocker, you can use `embed` to import it from a yaml file and leverage formatting